AdyenComponents/Boleto/BoletoComponent.swift

```diff
@@ -87,7 +87,15 @@ public final class BoletoComponent: PaymentComponent, LoadingComponent, Presenta
                                       apiContext: apiContext,
                                       fields: formFields,
                                       configuration: configuration,
-                                      onCreatePaymentDetails: { [weak self] in self?.createPaymentDetails() })
+                                      onCreatePaymentDetails: { [weak self] in
+            var paymentMethodDetails: PaymentMethodDetails?
+            do {
+                paymentMethodDetails = try self?.createPaymentDetails()
+            } catch let error {
+                adyenPrint(error)
```

Comment:
Hmm, would printing the error be enough? 🤔

Reply:
Not really. 
In addition - I am not sure that we can even test it. Since this scenario is rather a simple "Optional handling", then an actual business edge case..

Reply:
Yeah, seems we need some refactoring

Reply:
I thinK we can make the closure throw, right?